xknx/dpt/dpt_string.py

```diff
@@ -36,11 +36,11 @@ def to_knx(cls, value: str) -> tuple[int, ...]:
             knx_value = str(value)
             if not cls._test_boundaries(knx_value):
                 raise ValueError
-            raw = []
-            for character in knx_value:
-                raw.append(ord(character))
+            raw = [ord(character) for character in knx_value]
             raw.extend([0] * (cls.payload_length - len(raw)))
-            return tuple(raw)
+            # replace invalid characters with question marks
+            # bytes(knx_value, 'ascii') would raise UnicodeEncodeError
+            return tuple(map(lambda char: char if char <= 0xFF else ord("?"), raw))
```

Comment:
looks very cryptic on first sight, but there is a comment explaining it so I guess its fine